dbt/adapters/default/impl.py

```diff
@@ -151,6 +165,19 @@ def cancel_connection(self, connection):
     ###
     # FUNCTIONS THAT SHOULD BE ABSTRACT
     ###
+    def cache_new_relation(self, relation):
+        """Cache a new relation in dbt. It will show up in `list relations`."""
+        if relation is None:
+            dbt.exceptions.raise_compiler_error()
+        if dbt.flags.USE_CACHE:
+            self.cache.add(
+                schema=relation.schema,
```

Comment:
I think Connor indicated this above, but i think it would be wise to operate at a higher level of abstraction here. In the future, we're probably going to make it possible to configure the `database` (or `project`) that models get rendered into, and I imagine we won't want to go back and refactor caching when that change occurs.

Would it make sense to just pass in a `relation` here, and somehow make the Relation responsible for reporting it's own db/schema/identifier to the cache?